src/device_info.sh

```diff
@@ -676,6 +688,7 @@ function show_data()
 
   say 'CPU:'
   printf '  Model: %s\n' "${device_info_data['cpu_model']}"
+  printf '  Architecture: %s\n' "${device_info_data['cpu_architecture']}"
```

Comment:
The unit test for `show_data` will fail because you've added the architecture entry. To avoid this, in the test function `test_show_data` inside `device_test.sh` you need to accordingly add the line `Architecture: <some-arch>` in the `expected_cmd`, and a line `device_info_data['cpu_architecture']='<some-arch>`, like you did when you've added kernel info. Don't forget to do this in **this commit** (second).